compiler/rustc_codegen_llvm/src/back/write.rs

```diff
@@ -519,12 +519,22 @@ pub(crate) unsafe fn llvm_optimize(
     let pgo_sample_use_path = get_pgo_sample_use_path(config);
     let is_lto = opt_stage == llvm::OptStage::ThinLTO || opt_stage == llvm::OptStage::FatLTO;
     let instr_profile_output_path = get_instr_profile_output_path(config);
+    let sanitize_dataflow_abilist: Vec<_> = config
+        .sanitizer_dataflow_abilist
+        .iter()
+        .map(|file| CString::new(file.as_str()).unwrap())
+        .collect();
+    let sanitize_dataflow_abilist1: Vec<_> =
```

Comment:
I couldn't combine them into one variable because it seems the array of pointers don't live long enough until its use, so I had to bind it to an additional variable. I changed it to `sanitize_dataflow_abilist_ptrs` as suggested.